dev/TreeView/TreeViewItem.cpp

```diff
@@ -127,7 +127,7 @@ void TreeViewItem::OnDrop(winrt::DragEventArgs const& args)
 void TreeViewItem::OnDragOver(winrt::DragEventArgs const& args)
 {
     auto treeView = AncestorTreeView();
-    if (treeView)
+    if (treeView && !args.Handled())
```

Comment:
I wonder if this means we should prefer attaching to your own event over using c++ occlusion to "attach".  That way you wont even get called if the args are already handled.

Reply:
That's a good point, but I think there are some timing issues. The attached event always happens after the callback whereas in the callback override you have better control of when to call "super".

By looking at the code I think TreeView sometimes needs to take over the control and do stuff before calling "super" then set "handled" to true to block ListView's default behavior. If we switch to events in this case, ListView will do a bunch of stuff that we probably don't want before TreeView get a chance to handle it.